django/contrib/auth/hashers.py

```diff
@@ -211,6 +221,11 @@ def safe_summary(self, encoded):
         raise NotImplementedError('subclasses of BasePasswordHasher must provide a safe_summary() method')
 
     def must_update(self, encoded):
+        decoded = self.decode(encoded)
+        if 'salt' not in decoded:
+            warnings.warn(f'Method `decode` not fully implemented for {type(self).__name__}.\
+                            Decode will be required in future releases.',
+                          category=RemovedInDjango40Warning)
```

Comment:
`decode()` is a new method so I'm not sure if this warning has a lot of value, we want to add a new API and immediately deprecate it's unexpected implementation :shrug: .

IMO, we should rather document a new API (with release notes) and deprecate hashers without it, e.g.
```python
def decode(self):
    # RemovedInDjango41Warning: when the deprecation ends, replace with: NotImplementedError.
    warnings.warn(
        '%s.decode() method is not implemented. Using hashers without a '
        'decode() is deprecated and will no longer be supported in '
        'Django 4.1' % type(self).__name__,
        RemovedInDjango41Warning,
        stacklevel=2,
    )
    return {}
```

Reply:
This warning should be added to the `BasePasswordHasher` correct?